src/T2DMap.cpp

```diff
@@ -1325,7 +1325,8 @@ void T2DMap::paintEvent(QPaintEvent* e)
         auto font(painter.font());
         font.setPointSize(10);
         painter.setFont(font);
-        painter.drawText(0, 0, widgetWidth, widgetHeight, Qt::AlignCenter | Qt::TextWordWrap, tr("No map or no valid position."));
+        auto message = mpMap->mpRoomDB->size() == 0 ? tr("You don't have a map yet - load one, or start mapping from scratch to begin.") : tr("You have a map loaded (%1 rooms), but we don't know where you are at the moment.").arg(mpMap->mpRoomDB->size()); 
```

Comment:
Personally I do not like to see contractions such as "don't", "can't" etc. in computer messages - it keeps things a tiny bit clearer for translators if they have "do not", "can not" etc. - and the saving in space in English is not going to necessarily be reproducible in other languages anyhow.

Also for the second message, it is a prime candidate for a special (numeric or "numerous" as appears in the `mudlet.ts` file) translation.
```suggestion
        auto message = mpMap->mpRoomDB->size() == 0 ? tr("You do not have a map yet - load one, or start mapping from scratch to begin.") : tr("You have a map loaded (%n room(s)), but Mudlet does not know where you are at the moment.", mpMap->mpRoomDB->size())); 
```

N.B. That will require a manual update to the `en_US` translation so it will have 12 of them now! https://wiki.mudlet.org/w/Translating_Mudlet#Special_notes_on_separate_translations

Reply:
I agree with the numbers fix, but let's keep Mudlet informal and simple. It's a gaming app, not a super strict formal application!

Reply:
Sure, will try to update translations now.